src/Commands/PowerPlatform/Environment/GetPowerPlatformSolution.cs

```diff
@@ -0,0 +1,65 @@
+﻿using PnP.PowerShell.Commands.Base;
+using PnP.PowerShell.Commands.Utilities.REST;
+using System;
+using System.Management.Automation;
+using System.Linq;
+using PnP.PowerShell.Commands.Base.PipeBinds;
+
+namespace PnP.PowerShell.Commands.PowerPlatform.Environment
+{
+    [Cmdlet(VerbsCommon.Get, "PnPPowerPlatformSolution")]
+    public class GetPowerPlatformSolution: PnPAzureManagementApiCmdlet
+    {
+      
+        [Parameter(Mandatory = false, ValueFromPipeline = true)]
+        public PowerPlatformEnvironmentPipeBind Environment;
+
+        [Parameter(Mandatory = false)]
+        public PowerPlatformSolutionPipeBind Name;
+
+        protected override void ExecuteCmdlet()
+        {
+            string environmentName = null;
+            string dynamicsScopeUrl = null;
+            var environments = GraphHelper.GetResultCollectionAsync<Model.PowerPlatform.Environment.Environment>(Connection, "https://management.azure.com/providers/Microsoft.ProcessSimple/environments?api-version=2016-11-01", AccessToken).GetAwaiter().GetResult();
+            if (ParameterSpecified(nameof(Environment)))
+            {
+                environmentName = Environment.GetName().ToLower();
+                WriteVerbose($"Using environment as provided '{environmentName}'");
+                dynamicsScopeUrl = environments.FirstOrDefault(e => e.Properties.DisplayName.ToLower() == environmentName || e.Name.ToLower() == environmentName)?.Properties.LinkedEnvironmentMetadata.InstanceApiUrl;
+            }
+            else
+            {
+                environmentName = environments.FirstOrDefault(e => e.Properties.IsDefault.HasValue && e.Properties.IsDefault == true)?.Name;
+                dynamicsScopeUrl = environments.FirstOrDefault(e => e.Properties.IsDefault.HasValue && e.Properties.IsDefault == true)?.Properties.LinkedEnvironmentMetadata.InstanceApiUrl;
+                if (string.IsNullOrEmpty(environmentName))
+                {
+                    throw new Exception($"No default environment found, please pass in a specific environment name using the {nameof(Environment)} parameter");
+                }
+
+                WriteVerbose($"Using default environment as retrieved '{environmentName}'");
+            }
+
+            string accessTokenForGettingSolutions = TokenHandler.GetAccessTokenforPowerPlatformSolutions(this, Connection, dynamicsScopeUrl); 
```

Comment:
Can we use `RequiredMinimalApiPermissions` attribute instead of this ? Please check some graph cmdlets where we use it.

Reply:
This token that we are getting would have a dynamic URL based on the Power Platform environment....so token has to be get for url like "https://orge12323.crm8.dynamics.com" and this will change - so this would not worked with RequiredMinimalPermissions?  Let me know if you think there is any other way to get token